store/store.go

```
@@ -1,19 +1,164 @@
 package store
 
 import (
+	"context"
 	"sync"
 
+	"github.com/project-safari/zebra"
 	"github.com/project-safari/zebra/filestore"
 	"github.com/project-safari/zebra/idstore"
 	"github.com/project-safari/zebra/labelstore"
 	"github.com/project-safari/zebra/typestore"
 )
 
-//nolint:unused
 type ResourceStore struct {
 	lock sync.RWMutex
 	fs   *filestore.FileStore
 	ids  *idstore.IDStore
 	ls   *labelstore.LabelStore
 	ts   *typestore.TypeStore
 }
+
+func NewResourceStore(root string, factory zebra.ResourceFactory) *ResourceStore {
+	return &ResourceStore{
+		lock: sync.RWMutex{},
+		fs: func() *filestore.FileStore {
+			fs := filestore.NewFileStore(root, factory)
+			_ = fs.Initialize()
+
+			return fs
+		}(),
+		ids: nil,
+		ls:  nil,
+		ts:  nil,
+	}
+}
+
+func (rs *ResourceStore) Initialize() error {
+	resources, err := rs.fs.Load()
+	if err != nil {
+		return err
+	}
+
+	rs.ids = idstore.NewIDStore(resources)
+	rs.ls = labelstore.NewLabelStore(resources)
+	rs.ts = typestore.NewTypeStore(resources)
+
+	return nil
+}
+
+func (rs *ResourceStore) Wipe() error {
+	rs.fs = nil
+	rs.ids = nil
+	rs.ls = nil
+	rs.ts = nil
+
+	return nil
+}
+
+func (rs *ResourceStore) Clear() error {
+	_ = rs.fs.Clear()
```

you should return the error if any of these substores actually return an error